### PR TITLE
Documentation Improvements: update readme & add JSDOC

### PR DESCRIPTION
**Pull request by branchfractal**
**NOTES:**
- updated README
- added JSDoc style comments for various code 
- no logic changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,33 +9,37 @@ Official Haven Frontend Monorepository for Haven Desktop and Haven Web Version.
...
 
-`npm run start:web:mainnet`
+1. Navigate to client folder
+2. Build the app for the environment
+3. Run the appropriate script for the environment you want to run the app in
 
-For Testnet
+```bash
+cd client
+npm run build:web:testnet
```
Comment (MartyHav) [inline README.md:40]: Thanks or improving readme, it was insufficient.
One request:
npm run build:web:testnet will create a web build, 
for desktop build we would need to run: `npm run build:desktop:testnet` .
And for desktop development we need to start the client like: `npm run start:desktop:testnet`
And for the final Desktop build we need to run `sh ./sh/make_testnet.sh`
I think we can split this into own sections here for Develop and Build.

Comment (atreeleaf) [reply README.md]: updated, please let me know if there is anything else :)

Comment (branchfractal) [reply README.md]: updated

Comment (MartyHav) [reply README.md]: perfect

**Pull request merged**